### PR TITLE
TaskCompletionSource.TrySetCanceled fallback to overload without CancellationToken

### PR DESCRIPTION
**Pull request by OwnageIsMagic**
removed bugged netstandard implementation

If token is actually canceled
https://github.com/theraot/Theraot/commit/370eb542#diff-b111a79c94d34d7f7ad7d9914d8bb14f03673fd5726182d38e239ab55d1d2bb0L78-R77 introduced infinite recursion.
And if token is not canceled it will block until **the token** is canceled.

### CODE REVIEW

File: Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs [hunk 1]
```diff
@@ -28,64 +29,41 @@ private static class TrySetCanceledCachedDelegate<T>
 
             private static Func<TaskCompletionSource<T>, CancellationToken, bool> CreateTrySetCanceledDelegate()
             {
-                var trySetCanceledCached = typeof(TaskCompletionSource<T>).GetMethod
-                (
+                var trySetCanceledMethod = typeof(TaskCompletionSource<T>).GetMethod(
```
Comment (NN---) [inline Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs:32]: We do not need to do this every call.
Can we move it as static readonly field?

Comment (OwnageIsMagic) [reply Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs:32]: there is only 1 if from ??= on each call
https://github.com/theraot/Theraot/blob/7f93fd84606bf195220b3a870f6fa0e41077ee34/Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs#L25-L30

Comment (NN---) [reply Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs:32]: Ah . I see it is a lazy initialization.

Please mark unchanged variables as readonly so we don't change them.

**Review by NN--- — APPROVED** [review]

File: Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs (hunk 1 above)
Comment (OwnageIsMagic) [reply Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs:32]: It's lazy inited, so it can't be readonly

Comment (NN---) [reply Framework.Core/System/Threading/Tasks/TaskCompletionSourceTheraotExtensions.cs:32]: I see.

**Pull request merged**